From 540a5ac3eb4d987805e74a8b3d2d585186c2a0c4 Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdi@videolabs.io>
Date: Sun, 30 May 2021 21:22:40 +0200
Subject: [PATCH 05/77] contrib: gmp: add support for wasm-emscripten

---
 contrib/src/gmp/rules.mak | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/contrib/src/gmp/rules.mak b/contrib/src/gmp/rules.mak
index 11fe8acdfa..d6cd702ce0 100644
--- a/contrib/src/gmp/rules.mak
+++ b/contrib/src/gmp/rules.mak
@@ -13,6 +13,9 @@ ifeq ($(ARCH),mips64el)
 GMP_CONF += --disable-assembly
 endif
 endif
+ifdef HAVE_EMSCRIPTEN
+GMP_CONF += --disable-assembly
+endif
 
 ifdef HAVE_WIN32
 ifeq ($(ARCH),arm)
-- 
2.35.1

