From b28ea0ff21f0764f285f9094cd6f2359c91d9596 Mon Sep 17 00:00:00 2001
From: Etienne Brateau <etienne.brateau@gmail.com>
Date: Tue, 27 Apr 2021 16:19:40 +0200
Subject: [PATCH 16/77] vout: add emscripten window and webgl context

---
 include/vlc_vout_window.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/vlc_vout_window.h b/include/vlc_vout_window.h
index a3dfd1bfca..b5a01b33e4 100644
--- a/include/vlc_vout_window.h
+++ b/include/vlc_vout_window.h
@@ -65,6 +65,7 @@ enum vout_window_type {
     VOUT_WINDOW_TYPE_WAYLAND /**< Wayland surface */,
     VOUT_WINDOW_TYPE_DCOMP /**< Win32 DirectComposition */,
     VOUT_WINDOW_TYPE_KMS /**< DRM KMS CRTC */,
+    VOUT_WINDOW_TYPE_EMSCRIPTEN_WEBGL /**< Emscripten surface */,
 };
 
 /**
@@ -385,6 +386,7 @@ typedef struct vout_window_t {
         struct wl_surface *wl;   /**< Wayland surface (client pointer) */
         void     *dcomp_visual;  /**<  Win32 direct composition visual */
         uint32_t crtc;           /**< KMS CRTC identifier */
+        uint32_t em_context;     /* Emscripten webgl context */
     } handle;
 
     /** Display server (mandatory)
-- 
2.35.1

