From d36a5341abd7cbb4ad39af1dc9434ba1608a6678 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Thu, 3 Feb 2022 09:47:13 +0100
Subject: [PATCH 21/77] codec: Add a webcodec decoder device type

---
 include/vlc_codec.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/vlc_codec.h b/include/vlc_codec.h
index ff707e73a3..7460d03ed6 100644
--- a/include/vlc_codec.h
+++ b/include/vlc_codec.h
@@ -583,6 +583,7 @@ enum vlc_decoder_device_type
     VLC_DECODER_DEVICE_AWINDOW,
     VLC_DECODER_DEVICE_NVDEC,
     VLC_DECODER_DEVICE_MMAL,
+    VLC_DECODER_DEVICE_WEBCODEC,
 };
 
 struct vlc_decoder_device_operations
-- 
2.35.1

