From d80f3c5acc6455a4d3f8afd51b80d3f456203b85 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Thu, 3 Feb 2022 09:53:10 +0100
Subject: [PATCH 26/77] emscripten: build: Disable nvdec

---
 extras/package/wasm-emscripten/build.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/extras/package/wasm-emscripten/build.sh b/extras/package/wasm-emscripten/build.sh
index 5a1dbd230d..acd74591fc 100755
--- a/extras/package/wasm-emscripten/build.sh
+++ b/extras/package/wasm-emscripten/build.sh
@@ -168,6 +168,7 @@ if [ $BUILD_MODE -eq 1 ]; then
                         ac_cv_func_accept4=no \
                         ac_cv_func_pipe2=no \
                         --enable-gles2 \
+                        --disable-nvdec \
                         --with-contrib="$VLC_SRCPATH"/contrib/wasm32-unknown-emscripten
 fi
 
-- 
2.35.1

