From e8dffa4b13264baa22bdb5c59b3a6c638753a589 Mon Sep 17 00:00:00 2001
From: Alexandre Janniaux <ajanni@videolabs.io>
Date: Tue, 22 Feb 2022 17:45:12 +0100
Subject: [PATCH 35/77] opengl: mock: expose module on emscripten platform

---
 modules/video_output/opengl/Makefile.am | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules/video_output/opengl/Makefile.am b/modules/video_output/opengl/Makefile.am
index bc4079fefc..bdb5a1b3eb 100644
--- a/modules/video_output/opengl/Makefile.am
+++ b/modules/video_output/opengl/Makefile.am
@@ -108,6 +108,12 @@ libglfilter_mock_plugin_la_LIBADD += libvlc_opengl.la $(GL_LIBS)
 noinst_LTLIBRARIES += libglfilter_mock_plugin.la
 endif
 
+if HAVE_EMSCRIPTEN
+libglfilter_mock_plugin_la_LIBADD += libvlc_opengles.la $(GL_LIBS)
+libglfilter_mock_plugin_la_CFLAGS = -DUSE_OPENGL_ES2=1
+noinst_LTLIBRARIES += libglfilter_mock_plugin.la
+endif
+
 if HAVE_IOS
 libglfilter_mock_plugin_la_LIBADD += libvlc_opengles.la $(GLES2_LIBS)
 libglfilter_mock_plugin_la_CFLAGS = -DUSE_OPENGL_ES2=1
-- 
2.35.1

