From 098f9060b967025700bce2fe9e009e473f2eb0eb Mon Sep 17 00:00:00 2001
From: Alexandre Janniaux <ajanni@videolabs.io>
Date: Tue, 22 Feb 2022 17:55:36 +0100
Subject: [PATCH 38/77] opengl: mock: load API in mock

---
 modules/video_output/opengl/filter_mock.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules/video_output/opengl/filter_mock.c b/modules/video_output/opengl/filter_mock.c
index d455de4937..3915456d7e 100644
--- a/modules/video_output/opengl/filter_mock.c
+++ b/modules/video_output/opengl/filter_mock.c
@@ -102,6 +102,7 @@ struct sys {
 
     float rotation_matrix[16];
     float ar;
+    struct vlc_gl_api api;
 };
 
 static void
@@ -587,6 +588,9 @@ Open(struct vlc_gl_filter *filter, const config_chain_t *config,
 
     sys->sampler = NULL;
 
+    vlc_gl_api_Init(&sys->api, filter->gl);
+    filter->api = &sys->api;
+
     int ret;
     if (plane)
         ret = InitPlane(filter, glfmt);
-- 
2.35.1

