From 0cc2af768d3d7f01460b092bed2bdb37f72b8a46 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Mon, 9 May 2022 11:58:08 +0200
Subject: [PATCH 50/77] webcodec: Remove unused binding

---
 modules/codec/webcodec.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/modules/codec/webcodec.cpp b/modules/codec/webcodec.cpp
index 92b23d2b94..d44e9ee5ae 100644
--- a/modules/codec/webcodec.cpp
+++ b/modules/codec/webcodec.cpp
@@ -429,7 +429,3 @@ vlc_module_begin ()
     add_submodule()
         set_callback_dec_device(OpenDecDevice, 1)
 vlc_module_end ()
-
-EMSCRIPTEN_BINDINGS(vlc_picture) {
-    emscripten::class_<picture_t>("picture_t");
-}
-- 
2.35.1

