From 19e4b631ba7faf7e11bf2c07008346e6d5d3657a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Tue, 10 May 2022 10:31:16 +0200
Subject: [PATCH 56/77] webcodec: minor simplification

---
 modules/codec/webcodec.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/modules/codec/webcodec.cpp b/modules/codec/webcodec.cpp
index 51f63a2089..f82b1a7bf0 100644
--- a/modules/codec/webcodec.cpp
+++ b/modules/codec/webcodec.cpp
@@ -252,9 +252,7 @@ EM_JS(emscripten::EM_VAL, initDecoderJS, (void* decoder, emscripten::EM_VAL decC
 
         if (msg['customCmd'] == 'getDecoderWorkerMessagePort') {
             Module.msgChannel = new MessageChannel();
-            Module.msgChannel.port1.onmessage = function(msg) {
-                onDecoderWorkerMessage(msg);
-            };
+            Module.msgChannel.port1.onmessage = onDecoderWorkerMessage;
             self.postMessage({
                 customCmd: 'transferMessagePort',
                 targetThread: msg['replyTo'],
-- 
2.35.1

