From 85987521d4a5995038b155277268bbd07ab4c34d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Wed, 11 May 2022 12:03:39 +0200
Subject: [PATCH 58/77] webcodec: Fix invalid error return

---
 modules/codec/webcodec.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/codec/webcodec.cpp b/modules/codec/webcodec.cpp
index 4c6250294f..b3546edaea 100644
--- a/modules/codec/webcodec.cpp
+++ b/modules/codec/webcodec.cpp
@@ -400,7 +400,7 @@ static int Open( vlc_object_t* obj )
     {
         msg_Err( obj, "Can't get VideoDecoder type, webcodec is probably not "
                       "supported on this browser" );
-        return false;
+        return VLC_EGENERIC;
     }
     auto sys = std::make_unique<decoder_sys_t>();
     dec->p_sys = sys.get();
-- 
2.35.1

