From 935282fa99a0356589692c3493e143eccb77cd2d Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdi@videolabs.io>
Date: Mon, 8 Aug 2022 13:46:48 +0200
Subject: [PATCH 74/77] emjsfile: Close file access if open() failed

---
 modules/access/emjsfile.c | 25 +++++++++++++------------
 1 file changed, 13 insertions(+), 12 deletions(-)

diff --git a/modules/access/emjsfile.c b/modules/access/emjsfile.c
index 139b12b68a..488b6fb449 100644
--- a/modules/access/emjsfile.c
+++ b/modules/access/emjsfile.c
@@ -179,6 +179,14 @@ EM_ASYNC_JS(int, init_js_file, (stream_t *p_access, long id), {
     return return_value;
 });
 
+static void EmFileClose (vlc_object_t * p_this) {
+    stream_t *p_access = (stream_t*)p_this;
+    EM_ASM({
+            Module.vlcAccess[$0].worker_js_file = undefined;
+            Module.vlcAccess[$0].reader = undefined;            
+        }, p_access);
+}
+
 static int EmFileOpen( vlc_object_t *p_this ) {
     stream_t *p_access = (stream_t*)p_this;
 
@@ -247,6 +255,10 @@ static int EmFileOpen( vlc_object_t *p_this ) {
         return VLC_EGENERIC;
     }
 
+    access_sys_t *p_sys = vlc_obj_malloc(p_this, sizeof (*p_sys));
+    if (unlikely(p_sys == NULL))
+        return VLC_ENOMEM;
+    
     /*
       Request the file from the main thread.
       If it was not selected, it will return an error.
@@ -260,10 +272,6 @@ static int EmFileOpen( vlc_object_t *p_this ) {
         return VLC_EGENERIC;
     }
 
-    access_sys_t *p_sys = vlc_obj_malloc(p_this, sizeof (*p_sys));
-    if (unlikely(p_sys == NULL))
-        return VLC_ENOMEM;
-
     p_access->pf_read = Read;
     p_access->pf_block = NULL;
     p_access->pf_control = Control;
@@ -273,20 +281,13 @@ static int EmFileOpen( vlc_object_t *p_this ) {
     p_sys->offset = 0;
     if (get_js_file_size(p_access, &p_sys->js_file_size)) {
         msg_Err(p_access, "EMJsFile error: could not get file size!");
+        EmFileClose(p_this);
         return VLC_EGENERIC;
     } 
 
     return VLC_SUCCESS;
 }
 
-static void EmFileClose (vlc_object_t * p_this) {
-    stream_t *p_access = (stream_t*)p_this;
-    EM_ASM({
-            Module.vlcAccess[$0].worker_js_file = undefined;
-            Module.vlcAccess[$0].reader = undefined;            
-        }, p_access);
-}
-
 vlc_module_begin ()
     set_description( N_("Emscripten module to allow reading local files from the DOM's <input>") )
     set_shortname( N_("Emscripten Local File Input") )
-- 
2.35.1

