From 153eecd6159eaae8ab98dd98accc6eb7f033064d Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdi@videolabs.io>
Date: Mon, 8 Aug 2022 16:55:53 +0200
Subject: [PATCH 76/77] emjsfile: adapt to emscripten change

---
 modules/access/emjsfile.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/access/emjsfile.c b/modules/access/emjsfile.c
index 245a6a7201..9123a4f0fc 100644
--- a/modules/access/emjsfile.c
+++ b/modules/access/emjsfile.c
@@ -209,7 +209,7 @@ static int EmFileOpen( vlc_object_t *p_this ) {
     */
     MAIN_THREAD_EM_ASM({
         const thread_id = $0;
-        let w = Module.PThread.pthreads[thread_id].worker;
+        let w = Module.PThread.pthreads[thread_id];
         function handleFileRequest(e) {
             const msg = e.data;
             if (msg.type === "requestFile") {
-- 
2.35.1

